kivy/tests/test_urlrequest.py

```diff
@@ -5,17 +5,50 @@
 import pytest
 import threading
 
+from datetime import datetime
 from time import sleep
 from base64 import b64encode
 import os
 
+if not os.environ.get('NONETWORK'):
+    from kivy.network.urlrequest import UrlRequest
+    from kivy.clock import Clock
+
+
+def wait_request_is_finished(request, timeout=10):
+    start_time = datetime.now()
+    timed_out = False
+    while not request.is_finished and not timed_out:
+        Clock.tick()
+        sleep(.1)
+        timed_out = (datetime.now() - start_time).total_seconds() > timeout
+    assert request.is_finished
+
+
+def ensure_called_from_thread(queue):
+    """Ensures the callback is called from this thread (main)."""
+    tid = threading.get_ident()
+    assert queue[0][0] == tid
+    assert queue[-2][0] == tid
+    assert queue[-1][0] == tid
 
-class UrlRequestQueue(object):
 
-    queue = []
```

Comment:
The `queue` is already defined in the init, less is more. Having it defined here can lead to the misleading assumption we want it to be static attribute